proposed/2023/InsecureConnectionsDisableCertificateValidation.md

```diff
@@ -0,0 +1,135 @@
+- While the intention is a stopgap meeting where developers are today, it could promote bad security practices.
+
+## Rationale and alternatives
+
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+There are a number of ideas and proposals on the GitHub issues listed above. I have tried to combine them into simple and practical explanations with comparison to other tools adjacent to ours and their similar behaviors. I'm not diving into the specific scenarios for each of these behaviors, but rather just providing a high-level ability to turn off the checks that align with other tools.
+
+By default, every secure connection NuGet makes is verified to be secure before the transfer takes place. Using these options makes the transfer insecure.
+
+Here are two other alternatives that could be considered although they add a bit of a barrier to use.
+
+1. Make this feature user-specific for the nuget.config & properties. It is suggested there are a couple existing nuget.config features(proxy for example) that are only enabled if they are found in the scope of the `user` settings. See https://learn.microsoft.com/nuget/consume-packages/configuring-nuget-behavior#how-settings-are-applied for more details.
+2. Add detection for common local network sources. This would require a known list of these sources and ensuring it covers the common scenarios. We currently do not know what these are and would have to extend more customer development to learn. To my knowledge at the time of writing this, no other ecosystem does this.
+
+## Prior Art
+
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevant to this proposal? -->
+- curl has `--insecure` - https://curl.se/docs/manpage.html#-k
+- npm has `strict-ssl` - https://docs.npmjs.com/cli/v6/using-npm/config#strict-ssl
+- composer has `disable-tls` and `secure-http` - https://getcomposer.org/doc/06-config.md#disable-tls
+- maven wagon has a number of settings - https://maven.apache.org/wagon/wagon-providers/wagon-http/
+
+## Unresolved Questions
+
+<!-- What parts of the proposal do you expect to resolve before this gets accepted? -->
+<!-- What parts of the proposal need to be resolved before the proposal is stabilized? -->
+<!-- What related issues would you consider out of scope for this proposal but can be addressed in the future? -->
+- Are these the best descriptive names for these behaviors?
+    - Could they be simplified more?
+        - Other suggestions are: `DisableHttpsCertificateValidation`, `DisableTLSCertificateValidation`, and `DisableSSLCertificateValidation` (SSL is deprecated).
```

Comment:
I missed this, I like `DisableTLSCertificateValidation` best.

Reply:
Updated as this came up during team review too.